Bugfix/cam fixes 3

MarcelCaron-NOAA opened the pull request:
<b>Note to developers: You must use this PR template!</b>

## Description of Changes

> Please include a summary of the changes and the related GitHub issue(s). Please also include relevant motivation and context.

- _Bugfix:_ EVS Bugzilla 1547, i.e. all mpmd processes output to separate working directories, and output are merged to the main DATA directory after parallelization is complete.  _Outcome_: safer parallelization, and alignment with NCO code standards.
- _Bugfix_: a bug causing duplicate lines in final stats files.  _Outcome_: substantially reduced stats file sizes.
- _Bugfix_: incorrect HRRR TCDC field configuration. _Outcome_: an accurate TCDC stat.
- _Bugfix_: Various minor resource adjustments
- _Enhancement_: differentiated HRRR ASNOW and SNOD verification (already done in the upstream EVS-RRFS release branch)
- _Enhancement_: added CAPE and ASNOW graphics
- _Enhancement_: adjusted 3-h Precipitation and 6-hr Snowfall (WEASD, SNOD, ASNOW) graphics so that 24-hour periods are more accurately covered, and extended forecast verification is limited to models with leads out to F60.


## Developer Questions and Checklist
* Is this a high priority PR? If so, why and is there a date it needs to be merged by?

No
* Do you have any planned upcoming annual leave/PTO?

No
* Are there any changes needed for when the jobs are supposed to run?

No
  
- [x] The code changes follow [NCO's EE2 Standards](https://www.nco.ncep.noaa.gov/idsb/implementation_standards/ImplementationStandards.v11.0.0.pdf).
- [x] Developer's name is removed throughout the code and have used `${USER}` where necessary throughout the code.
- [x] References the feature branch for `HOMEevs` are removed from the code.
- [x] J-Job environment variables, COMIN and COMOUT directories, and output follow what has been [defined](https://docs.google.com/document/d/1JWg_4q80aYmmAoD21GFjp9R9y5-3w7WGM3-0HJk0Pjs/edit#heading=h.7ysbr191vzu4) for EVS.
- [x] Jobs over 15 minutes in runtime have restart capability.
- [x] If applicable, changes in the `dev/drivers/scripts` or `dev/modulefiles` have been made in the corresponding `ecf/scripts` and `ecf/defs/evs-nco.def`? 
- [x] Jobs contain the approriate file checking and don't run METplus for any missing data.
- [x] Code is using METplus wrappers structure and not calling MET executables directly.
- [x] Log is free of any ERRORs or WARNINGs.

## Testing Instructions

> Please include testing instructions for the PR assignee. Include all relevant input datasets needed to run the tests.

#### (1) Set up jobs

- symlink the EVS_fix directory locally as "fix"
- In each driver script, edit the following environment variables:
**HOMEevs** - set to your test EVS directory
**COMIN** - set to `/lfs/h2/emc/vpppg/noscrub/emc.vpppg/${NET}_beta5/$evs_ver_2d`
**COMOUT** - set to your test output directory
**SENDMAIL** - (_optional_) set to `"NO"`
**DATAROOT** - (_optional_) set to your test DATAROOT directory

#### (2) Running jobs

I recommend testing the following jobs:
```
jevs_cam_hireswarw_severe_prep
jevs_cam_hireswarwmem2_severe_prep
jevs_cam_hireswfv3_severe_prep
jevs_cam_hrrr_severe_prep
jevs_cam_namnest_severe_prep
jevs_cam_hireswarw_precip_prep
jevs_cam_hireswarwmem2_precip_prep
jevs_cam_hireswfv3_precip_prep
jevs_cam_hrrr_precip_prep
jevs_cam_namnest_precip_prep
```
```
jevs_cam_hireswarw_grid2obs_stats
jevs_cam_hireswarwmem2_grid2obs_stats
jevs_cam_hireswfv3_grid2obs_stats
jevs_cam_hrrr_grid2obs_stats
jevs_cam_namnest_grid2obs_stats
jevs_cam_hireswarw_precip_stats
jevs_cam_hireswarwmem2_precip_stats
jevs_cam_hireswfv3_precip_stats
jevs_cam_hrrr_precip_stats
jevs_cam_namnest_precip_stats
jevs_cam_hireswarw_snowfall_stats
jevs_cam_hireswarwmem2_snowfall_stats
jevs_cam_hireswfv3_snowfall_stats
jevs_cam_hrrr_snowfall_stats
jevs_cam_namnest_snowfall_stats
jevs_cam_hireswarw_radar_stats
jevs_cam_hireswarwmem2_radar_stats
jevs_cam_hireswfv3_radar_stats
jevs_cam_hrrr_radar_stats
jevs_cam_namnest_radar_stats
jevs_cam_hireswarw_severe_stats
jevs_cam_hireswarwmem2_severe_stats
jevs_cam_hireswfv3_severe_stats
jevs_cam_hrrr_severe_stats
jevs_cam_href_severe_stats
jevs_cam_namnest_severe_stats
```
```
jevs_cam_headline_plots
jevs_cam_grid2obs_last31days_plots
jevs_cam_grid2obs_last90days_plots
jevs_cam_precip_last31days_plots
jevs_cam_precip_last90days_plots
jevs_cam_snowfall_plots
```
[Total: 10 prep jobs; 26 stats jobs; 6 plots jobs]

- All _precip stats_ driver scripts can be submitted using `qsub -v vhr=21,VDATE=$(date -d "-3 days" +"%Y%m%d") $driver`
- All other driver scripts can be submitted using `qsub -v vhr=00 $driver`
- All jobs can be submitted at any time

#### (3) Checking jobs
- Log files should be checked by the developer for the following keywords:
```
check="FATAL\|WARNING\|error\|Killed\|Cgroup\|argument expected\|No such file\|cannot\|failed\|unexpected\|exceeded"
grep "$check" $logfile
```

Commit:
Fix Look-in dir for gather3

Commit:
Bugzilla 1547: Run child processes in separate working directories

Commit:
Bugzilla 1547: Run child processes in separate working directories
(severe and radar plots jobs)

Commit:
Fix ASNOW verification + adjust snowfall graphics

Commit:
Adjust precip graphics

Commit:
Fix HRRR TCDC field configuration

Commit:
Add CONUS E/W/C/S graphics

Commit:
update walltime

Commit:
Adjust severe prep resource allocation + match ecf with dev

Commit:
Fix input template for merged ptype file

Commit:
Fix metric config

Commit:
Merge branch 'develop' into bugfix/cam_fixes_3

malloryprow commented on scripts/plots/cam/exevs_cam_grid2obs_plots.sh:
```diff
@@ -71,6 +75,12 @@ else
 
+# Copy Plots Output to Main Directory
+for CHILD_DIR in ${DATA}/${VERIF_CASE}/out/workdirs/*; do
+    cp -ru $CHILD_DIR/* ${DATA}/${VERIF_CASE}/out/.
```
Could we do `cp -ruv`? The `-v` is the verbose option.

malloryprow commented on scripts/plots/cam/exevs_cam_headline_plots.sh:
```diff
@@ -70,6 +74,12 @@ else
 
+# Copy Plots Output to Main Directory
+for CHILD_DIR in ${DATA}/${VERIF_CASE}/out/workdirs/*; do
+    cp -ru $CHILD_DIR/* ${DATA}/${VERIF_CASE}/out/.
```
Could we do `cp -ruv`? The `-v` is the verbose option.

malloryprow commented on scripts/plots/cam/exevs_cam_radar_plots.sh:
```diff
@@ -177,6 +175,15 @@ else
+# Copy Plots Output to Main Directory
+###################################################################
+for CHILD_DIR in ${DATA}/out/workdirs/*; do
+    cp -ru $CHILD_DIR/* ${DATA}/out/.
```
Could we do `cp -ruv`? The `-v` is the verbose option.

malloryprow commented on scripts/plots/cam/exevs_cam_severe_plots.sh:
```diff
@@ -191,6 +189,15 @@ else
+# Copy Plots Output to Main Directory
+###################################################################
+for CHILD_DIR in ${DATA}/out/workdirs/*; do
+    cp -ru $CHILD_DIR/* ${DATA}/out/.
```
Could we do `cp -ruv`? The `-v` is the verbose option.

malloryprow commented on scripts/plots/cam/exevs_cam_snowfall_plots.sh:
```diff
@@ -70,6 +74,12 @@ else
 
+# Copy Plots Output to Main Directory
+for CHILD_DIR in ${DATA}/${VERIF_CASE}/out/workdirs/*; do
+    cp -ru $CHILD_DIR/* ${DATA}/${VERIF_CASE}/out/.
```
Could we do `cp -ruv`? The `-v` is the verbose option.

malloryprow commented on scripts/stats/cam/exevs_cam_radar_stats.sh:
```diff
@@ -106,6 +107,15 @@ else
+# Copy child output to main output directory
+###################################################################
+for CHILD_DIR in ${DATA}/workdirs/*; do
+    cp -ru ${CHILD_DIR}/* ${DATA}/.
```
Could we do `cp -ruv`? The `-v` is the verbose option.

malloryprow commented on scripts/stats/cam/exevs_hireswarw_grid2obs_stats.sh:
```diff
@@ -110,6 +114,12 @@ else
 
+# Copy Reformat Output to Main Directory
+for CHILD_DIR in ${DATA}/${VERIF_CASE}/METplus_output/workdirs/${job_type}/*; do
+    cp -ru $CHILD_DIR/* ${DATA}/${VERIF_CASE}/METplus_output/.
```
Could we do `cp -ruv`? The `-v` is the verbose option.

malloryprow commented on scripts/stats/cam/exevs_hireswarw_grid2obs_stats.sh:
```diff
@@ -179,6 +193,12 @@ else
 
+# Copy Generate Output to Main Directory
+for CHILD_DIR in ${DATA}/${VERIF_CASE}/METplus_output/workdirs/${job_type}/*; do
+    cp -ru $CHILD_DIR/* ${DATA}/${VERIF_CASE}/METplus_output/.
```
Could we do `cp -ruv`? The `-v` is the verbose option.

malloryprow commented on scripts/stats/cam/exevs_hireswarw_grid2obs_stats.sh:
```diff
@@ -237,6 +261,12 @@ else
 
+# Copy Gather Output to Main Directory
+for CHILD_DIR in ${DATA}/${VERIF_CASE}/METplus_output/workdirs/${job_type}/*; do
+    cp -ru $CHILD_DIR/* ${DATA}/${VERIF_CASE}/METplus_output/.
```
Could we do `cp -ruv`? The `-v` is the verbose option.

malloryprow commented on scripts/stats/cam/exevs_hireswarw_grid2obs_stats.sh:
```diff
@@ -292,13 +326,19 @@ else
 
+# Copy Gather 2 Output to Main Directory
+for CHILD_DIR in ${DATA}/${VERIF_CASE}/METplus_output/workdirs/${job_type}/*; do
+    cp -ru $CHILD_DIR/* ${DATA}/${VERIF_CASE}/METplus_output/.
```
Could we do `cp -ruv`? The `-v` is the verbose option.

malloryprow commented on scripts/stats/cam/exevs_hireswarw_grid2obs_stats.sh:
```diff
@@ -358,5 +402,11 @@ if [ "$vhr" -ge "$last_cyc" ]; then
+
+        # Copy Gather 3 Output to Main Directory
+        for CHILD_DIR in ${DATA}/${VERIF_CASE}/METplus_output/workdirs/${job_type}/*; do
+            cp -ru $CHILD_DIR/* ${DATA}/${VERIF_CASE}/METplus_output/.
```
Could we do `cp -ruv`? The `-v` is the verbose option.

malloryprow commented on scripts/stats/cam/exevs_hireswarw_precip_stats.sh:
```diff
@@ -115,6 +119,12 @@ else
 
+# Copy Reformat Output to Main Directory
+for CHILD_DIR in ${DATA}/${VERIF_CASE}/METplus_output/workdirs/${job_type}/*; do
+    cp -ru $CHILD_DIR/* ${DATA}/${VERIF_CASE}/METplus_output/.
```
Could we do `cp -ruv`? The `-v` is the verbose option.

malloryprow commented on scripts/stats/cam/exevs_hireswarw_precip_stats.sh:
```diff
@@ -195,6 +209,12 @@ else
 
+# Copy Generate Output to Main Directory
+for CHILD_DIR in ${DATA}/${VERIF_CASE}/METplus_output/workdirs/${job_type}/*; do
+    cp -ru $CHILD_DIR/* ${DATA}/${VERIF_CASE}/METplus_output/.
```
Could we do `cp -ruv`? The `-v` is the verbose option.

malloryprow commented on scripts/stats/cam/exevs_hireswarw_precip_stats.sh:
```diff
@@ -249,6 +273,12 @@ else
 
+# Copy Gather Output to Main Directory
+for CHILD_DIR in ${DATA}/${VERIF_CASE}/METplus_output/workdirs/${job_type}/*; do
+    cp -ru $CHILD_DIR/* ${DATA}/${VERIF_CASE}/METplus_output/.
```
Could we do `cp -ruv`? The `-v` is the verbose option.

malloryprow commented on scripts/stats/cam/exevs_hireswarw_precip_stats.sh:
```diff
@@ -301,13 +335,19 @@ else
 
+# Copy Gather 2 Output to Main Directory
+for CHILD_DIR in ${DATA}/${VERIF_CASE}/METplus_output/workdirs/${job_type}/*; do
+    cp -ru $CHILD_DIR/* ${DATA}/${VERIF_CASE}/METplus_output/.
```
Could we do `cp -ruv`? The `-v` is the verbose option.

malloryprow commented on scripts/stats/cam/exevs_hireswarw_precip_stats.sh:
```diff
@@ -377,5 +421,11 @@ if [ "$vhr" -ge "$last_cyc" ]; then
+
+        # Copy Gather 3 Output to Main Directory
+        for CHILD_DIR in ${DATA}/${VERIF_CASE}/METplus_output/workdirs/${job_type}/*; do
+            cp -ru $CHILD_DIR/* ${DATA}/${VERIF_CASE}/METplus_output/.
```
Could we do `cp -ruv`? The `-v` is the verbose option.

malloryprow commented:
I started commenting on the lines but realized it was in quite a few places so leaving a general comment here for a change.

In the ex-scripts, there is the copying of the child directories output to a main output directory using `cp -ru`. Could we do `cp -ruv`? The `-v` is the verbose option.

malloryprow commented:
I see the testing for the prep involves the severe dev drivers but nothing for the precip. The precip prep jobs are suspected jobs using MPMD, which this PR says it is addressing (https://github.com/NOAA-EMC/EVS/issues/551). Are these jobs using MPMD?

AndrewBenjamin-NOAA commented on scripts/plots/cam/exevs_cam_grid2obs_plots.sh:
```diff
@@ -88,7 +98,7 @@ if [ -d $log_dir ]; then
 
 # Tar and Copy output files to EVS COMOUT directory
-find ${DATA}/${VERIF_CASE}/* -name "*.png" -type f -print | tar -cvf ${DATA}/${NET}.${STEP}.${COMPONENT}.${RUN}.${VERIF_CASE}.${EVAL_PERIOD}.v${VDATE}.tar --transform='s#.*/##' -T -
+find ${DATA}/${VERIF_CASE}/* -name "*.png" -type f -not -path "*workdirs*" -print | tar -cvf ${DATA}/${NET}.${STEP}.${COMPONENT}.${RUN}.${VERIF_CASE}.${EVAL_PERIOD}.v${VDATE}.tar --transform='s#.*/##' -T -
```
@MarcelCaron-NOAA is this to avoid copying the plots twice? Once from the Main directory and once from the working dir?

MarcelCaron-NOAA replied:
exactly, otherwise I was getting tons of duplicate plots in the final tar file!

MarcelCaron-NOAA commented:
> I see the testing for the prep involves the severe dev drivers but nothing for the precip. The precip prep jobs are suspected jobs using MPMD, which this PR says it is addressing (#551). Are these jobs using MPMD?

@malloryprow You're right, and yes these jobs use MPMD.  I must have missed them when grepping for the mpi command.  I'll add and test the fix for those five jobs and then update this thread.

Commit:
Use verbose option for cp commands

MarcelCaron-NOAA commented:
> I started commenting on the lines but realized it was in quite a few places so leaving a general comment here for a change.
> 
> In the ex-scripts, there is the copying of the child directories output to a main output directory using `cp -ru`. Could we do `cp -ruv`? The `-v` is the verbose option.

Yes, just added that change in https://github.com/NOAA-EMC/EVS/pull/597/commits/01928c7467ee72a20dbd2ba2d678a46764536014

malloryprow commented:
Thank you!

Commit:
Bugzilla 1547: Run child processes in separate working directories
(precip prep jobs)

MarcelCaron-NOAA commented:
@malloryprow Okay!  The cam precip prep jobs have the MPMD fix and were tested for successful completion.  The PR testing instructions were updated to include the new jobs.

malloryprow commented:
### prep - severe

### jevs_cam_hireswarw_severe_prep
**Log File:** /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr597/EVS/dev/drivers/scripts/prep/cam/jevs_cam_hireswarw_severe_prep.o159672091
**COMOUT:** /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr597/evs/v2.0/prep/cam
**DATA:** /lfs/h2/emc/stmp/mallory.row/evs_test/prod/tmp/jevs_cam_hireswarw_severe_prep.159672091.cbqs01

### jevs_cam_hireswarwmem2_severe_prep
**Log File:** /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr597/EVS/dev/drivers/scripts/prep/cam/jevs_cam_hireswarwmem2_severe_prep.o159672069
**COMOUT:** /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr597/evs/v2.0/prep/cam
**DATA:** /lfs/h2/emc/stmp/mallory.row/evs_test/prod/tmp/jevs_cam_hireswarwmem2_severe_prep.159672069.cbqs01

### jevs_cam_hireswfv3_severe_prep
**Log File:** /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr597/EVS/dev/drivers/scripts/prep/cam/jevs_cam_hireswfv3_severe_prep.o159672096
**COMOUT:** /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr597/evs/v2.0/prep/cam
**DATA:** /lfs/h2/emc/stmp/mallory.row/evs_test/prod/tmp/jevs_cam_hireswfv3_severe_prep.159672096.cbqs01

### jevs_cam_hrrr_severe_prep
**Log File:** /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr597/EVS/dev/drivers/scripts/prep/cam/jevs_cam_hrrr_severe_prep.o159672098
**COMOUT:** /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr597/evs/v2.0/prep/cam/
**DATA:** /lfs/h2/emc/stmp/mallory.row/evs_test/prod/tmp/jevs_cam_hrrr_severe_prep.159672098.cbqs01

### jevs_cam_namnest_severe_prep
**Log File:** /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr597/EVS/dev/drivers/scripts/prep/cam/jevs_cam_namnest_severe_prep.o159672100
**COMOUT:** /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr597/evs/v2.0/prep/cam
**DATA:** /lfs/h2/emc/stmp/mallory.row/evs_test/prod/tmp/jevs_cam_namnest_severe_prep.159672100.cbqs01

MarcelCaron-NOAA commented:
severe prep jobs completed cleanly and output files look normal!
✔️ jevs_cam_hireswarw_severe_prep
✔️ jevs_cam_hireswarwmem2_severe_prep
✔️ jevs_cam_hireswfv3_severe_prep
✔️ jevs_cam_hrrr_severe_prep
✔️ jevs_cam_namnest_severe_prep

malloryprow commented:
Awesome, I'll get the precip jobs set up to test and then run. Stay tuned!

malloryprow commented:
### prep - precip - Try 1

### jevs_cam_hireswarw_precip_prep
**Log File:** /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr597/EVS/dev/drivers/scripts/prep/cam/jevs_cam_hireswarw_precip_prep.o159682031
**COMOUT:** /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr597/evs/v2.0/prep/cam
**DATA:** /lfs/h2/emc/stmp/mallory.row/evs_test/prod/tmp/jevs_cam_hireswarw_precip_prep.159682031.cbqs01

### jevs_cam_hireswarwmem2_precip_prep
**Log File:** /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr597/EVS/dev/drivers/scripts/prep/cam/jevs_cam_hireswarwmem2_precip_prep.o159682065 
**COMOUT:** /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr597/evs/v2.0/prep/cam
**DATA:** /lfs/h2/emc/stmp/mallory.row/evs_test/prod/tmp/jevs_cam_hireswarwmem2_precip_prep.159682065.cbqs01

### jevs_cam_hireswfv3_precip_prep
**Log File:** /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr597/EVS/dev/drivers/scripts/prep/cam/jevs_cam_hireswfv3_precip_prep.o159682082
**COMOUT:** /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr597/evs/v2.0/prep/cam
**DATA:** /lfs/h2/emc/stmp/mallory.row/evs_test/prod/tmp/jevs_cam_hireswfv3_precip_prep.159682082.cbqs01

### jevs_cam_hrrr_precip_prep
**Log File:** /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr597/EVS/dev/drivers/scripts/prep/cam/jevs_cam_hrrr_precip_prep.o159682137
**COMOUT:** /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr597/evs/v2.0/prep/cam
**DATA:** /lfs/h2/emc/stmp/mallory.row/evs_test/prod/tmp/jevs_cam_hrrr_precip_prep.159682137.cbqs01

### jevs_cam_namnest_precip_prep
**Log File:** /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr597/EVS/dev/drivers/scripts/prep/cam/jevs_cam_namnest_precip_prep.o159682307
**COMOUT:** /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr597/evs/v2.0/prep/cam
**DATA:** /lfs/h2/emc/stmp/mallory.row/evs_test/prod/tmp/jevs_cam_namnest_precip_prep.159682307.cbqs01

malloryprow commented:
So it appears that the jobs are not using CFP? I see `USE_CFP=NO`, but the resources in the drivers are all requesting multiple CPUS (`select=1:ncpus=128`). The infrastructure is there to support MPMD in these jobs, though.

How should these jobs be run? If `USE_CFP=NO`, it is running serially and they don't need the individual job directories. The resources should be set to `select=1:ncpus=1`.

Commit:
use cfp + reduce resources

MarcelCaron-NOAA commented:
Oh interesting.  Thanks for catching that.  Let's go with USE_CFP=YES.  I just tested that and it worked.  I also pushed a change to reduce the ncpus (128 is way too many).  

I don't remember why USE_CFP was set to NO.  My guess is that individual processes were stepping on each other during copy commands (main purpose of this job).  With the Bugzilla fix that can't happen, and so this can be set to YES.

malloryprow commented:
### prep - precip - Try 2

### jevs_cam_hireswarw_precip_prep
**Log File:** /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr597/EVS/dev/drivers/scripts/prep/cam/jevs_cam_hireswarw_precip_prep.o159697253
**COMOUT:** /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr597/evs/v2.0/prep/cam
**DATA:** /lfs/h2/emc/stmp/mallory.row/evs_test/prod/tmp/jevs_cam_hireswarw_precip_prep.159697253.cbqs01

### jevs_cam_hireswarwmem2_precip_prep
**Log File:** /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr597/EVS/dev/drivers/scripts/prep/cam/jevs_cam_hireswarwmem2_precip_prep.o159697258
**COMOUT:** /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr597/evs/v2.0/prep/cam
**DATA:** /lfs/h2/emc/stmp/mallory.row/evs_test/prod/tmp/jevs_cam_hireswarwmem2_precip_prep.159697258.cbqs01

### jevs_cam_hireswfv3_precip_prep
**Log File:** /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr597/EVS/dev/drivers/scripts/prep/cam/jevs_cam_hireswfv3_precip_prep.o159697291
**COMOUT:** /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr597/evs/v2.0/prep/cam
**DATA:** /lfs/h2/emc/stmp/mallory.row/evs_test/prod/tmp/jevs_cam_hireswfv3_precip_prep.159697291.cbqs01

### jevs_cam_hrrr_precip_prep
**Log File:** /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr597/EVS/dev/drivers/scripts/prep/cam/jevs_cam_hrrr_precip_prep.o159697315
**COMOUT:** /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr597/evs/v2.0/prep/cam
**DATA:** /lfs/h2/emc/stmp/mallory.row/evs_test/prod/tmp/jevs_cam_hrrr_precip_prep.159697315.cbqs01

### jevs_cam_namnest_precip_prep
**Log File:** /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr597/EVS/dev/drivers/scripts/prep/cam/jevs_cam_namnest_precip_prep.o159697363
**COMOUT:** /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr597/evs/v2.0/prep/cam
**DATA:** /lfs/h2/emc/stmp/mallory.row/evs_test/prod/tmp/jevs_cam_namnest_precip_prep.159697363.cbqs01

MarcelCaron-NOAA commented:
All precip prep jobs completed cleanly and output look normal:
✔️ jevs_cam_hireswarw_precip_prep
✔️ jevs_cam_hireswarwmem2_precip_prep
✔️ jevs_cam_hireswfv3_precip_prep
✔️ jevs_cam_hrrr_precip_prep
✔️ jevs_cam_namnest_precip_prep

👍 no other concerns from me about **prep** jobs

malloryprow commented:
Awesome!

One thing I noticed was a lot of empty directories in the workdirs directory. `/lfs/h2/emc/stmp/mallory.row/evs_test/prod/tmp/jevs_cam_hireswarw_precip_prep.159697253.cbqs01/precip/data/workdirs/job*/*`

Is it possible to make the directories underneath workdirs/job* only be created if they are going to be used?

MarcelCaron-NOAA commented:
I have to think about that.  The logic I used is to replicate the "main" directory in each working directory (workdirs/job*), then after MPMD copy any outfiles to the main directory--so naturally some working directories are not used, but all main directories are used.  I don't write each working directory specifically for its output.  The stats and plots components will have the same issue.  Will NCO flag us if working directories have empty directories?

AliciaBentley-NOAA commented:
@malloryprow This might be a question worth asking Steven Earle and Wei Wei. I know that the EE2 Review process flagged a lot of empty directories, but does it matter if they are inside the working directory itself? Might be worth clarifying as we address the mpmd bugzilla! 

CC @MarcelCaron-NOAA @AndrewBenjamin-NOAA

malloryprow commented:
Sounds good! Will revive that email and ask!

AliciaBentley-NOAA commented:
@MarcelCaron-NOAA Mallory may have told you this offline, but I wanted to add it to the discussion for future reference. 

@malloryprow asked NCO:

> ...We have a developer working on this Bugzilla for one of the components. Their thought process was for each job to have workdirs/job[some number] and within each of those to recreate the "main" directories. This results in some of those directories being empty (not every job makes use of the directory within its worksdirs/job[some number]). Does not having a lot of empty directories also apply to the MPMD job working directories?

And NCO (Wei Wei) said:

> The empty directory is more of a concern for $COM than $DATA. If the question is specified to $DATA/job*, I would say it's fine as is.

So I think you're good to go with this, but let us know if that's not how you take this information!

MarcelCaron-NOAA commented:
@AliciaBentley-NOAA @malloryprow Okay!  Thanks for asking Wei!  It sounds like the empty directories are fine but not preferred. 
 If it works for you I'll leave the current logic in this PR just to get the bugzilla fix through, then put in a change in a future PR addressing the empty directories.

malloryprow commented:
Yup, fine with me. I'll get testing the grid2obs stats next here today, hopefully before lunch.

malloryprow commented:
### stats - grid2obs - Try 1

Submitted all jobs with vhr=00. COMOUT is /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr597/evs/v2.0/stats/cam.

### jevs_cam_hireswarw_grid2obs_stats
**Log File:** /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr597/EVS/dev/drivers/scripts/stats/cam/jevs_cam_hireswarw_grid2obs_stats.o159797962
**DATA:** /lfs/h2/emc/stmp/mallory.row/evs_test/prod/tmp/jevs_cam_hireswarw_grid2obs_stats.159797962.cbqs01

### jevs_cam_hireswarwmem2_grid2obs_stats
**Log File:** /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr597/EVS/dev/drivers/scripts/stats/cam/jevs_cam_hireswarwmem2_grid2obs_stats.o159797984
**DATA:** /lfs/h2/emc/stmp/mallory.row/evs_test/prod/tmp/jevs_cam_hireswarwmem2_grid2obs_stats.159797984.cbqs01

### jevs_cam_hireswfv3_grid2obs_stats
**Log File:** /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr597/EVS/dev/drivers/scripts/stats/cam/jevs_cam_hireswfv3_grid2obs_stats.o159798016
**DATA:** /lfs/h2/emc/stmp/mallory.row/evs_test/prod/tmp/jevs_cam_hireswfv3_grid2obs_stats.159798016.cbqs01

### jevs_cam_hrrr_grid2obs_stats
**Log File:** /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr597/EVS/dev/drivers/scripts/stats/cam/jevs_cam_hrrr_grid2obs_stats.o159798022
**DATA:** /lfs/h2/emc/stmp/mallory.row/evs_test/prod/tmp/jevs_cam_hrrr_grid2obs_stats.159798022.cbqs01

### jevs_cam_namnest_grid2obs_stats
**Log File:** /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr597/EVS/dev/drivers/scripts/stats/cam/jevs_cam_namnest_grid2obs_stats.o159798153
**DATA:** /lfs/h2/emc/stmp/mallory.row/evs_test/prod/tmp/jevs_cam_namnest_grid2obs_stats.159798153.cbqs01

MarcelCaron-NOAA commented:
@malloryprow The following jobs completed cleanly and output look normal:
✔️ jevs_cam_hireswarw_grid2obs_stats
✔️ jevs_cam_hireswarwmem2_grid2obs_stats
✔️ jevs_cam_hireswfv3_grid2obs_stats

The following completed with errors:
❌ jevs_cam_hrrr_grid2obs_stats
❌ jevs_cam_namnest_grid2obs_stats

I have a fix, but just running it for both jobs myself first, then I'll push it shortly!

Commit:
Look in main output dir for firewx mask input

MarcelCaron-NOAA commented:
@malloryprow Alright, the remaining two grid2obs stats jobs should be good to go when you get a chance!

malloryprow commented:
### stats - grid2obs - Try 1

Just re-ran all of them. Submitted all jobs with vhr=00. COMOUT is /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr597/evs/v2.0/stats/cam.

### jevs_cam_hireswarw_grid2obs_stats
**Log File:** /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr597/EVS/dev/drivers/scripts/stats/cam/jevs_cam_hireswarw_grid2obs_stats.o159890259
**DATA:** /lfs/h2/emc/stmp/mallory.row/evs_test/prod/tmp/jevs_cam_hireswarw_grid2obs_stats.159890259.cbqs01

### jevs_cam_hireswarwmem2_grid2obs_stats
**Log File:** /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr597/EVS/dev/drivers/scripts/stats/cam/jevs_cam_hireswarwmem2_grid2obs_stats.o159890270
**DATA:** /lfs/h2/emc/stmp/mallory.row/evs_test/prod/tmp/jevs_cam_hireswarwmem2_grid2obs_stats.159890270.cbqs01

### jevs_cam_hireswfv3_grid2obs_stats
**Log File:** /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr597/EVS/dev/drivers/scripts/stats/cam/jevs_cam_hireswfv3_grid2obs_stats.o159890297
**DATA:** /lfs/h2/emc/stmp/mallory.row/evs_test/prod/tmp/jevs_cam_hireswfv3_grid2obs_stats.159890297.cbqs01

### jevs_cam_hrrr_grid2obs_stats
**Log File:** /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr597/EVS/dev/drivers/scripts/stats/cam/jevs_cam_hrrr_grid2obs_stats.o159890311
**DATA:** /lfs/h2/emc/stmp/mallory.row/evs_test/prod/tmp/jevs_cam_hrrr_grid2obs_stats.159890311.cbqs01

### jevs_cam_namnest_grid2obs_stats
**Log File:** /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr597/EVS/dev/drivers/scripts/stats/cam/jevs_cam_namnest_grid2obs_stats.o159890321
**DATA:** /lfs/h2/emc/stmp/mallory.row/evs_test/prod/tmp/jevs_cam_namnest_grid2obs_stats.159890321.cbqs01

MarcelCaron-NOAA commented:
Thanks @malloryprow!  The following completed cleanly, data directory and output files look normal:
✔️ jevs_cam_hireswarw_grid2obs_stats
✔️ jevs_cam_hireswarwmem2_grid2obs_stats
✔️ jevs_cam_hireswfv3_grid2obs_stats
✔️ jevs_cam_namnest_grid2obs_stats

Still running:
⏳ jevs_cam_hrrr_grid2obs_stats

MarcelCaron-NOAA commented:
✔️ jevs_cam_hrrr_grid2obs_stats completed cleanly and output looks good!

That should be all five grid2obs stats jobs

malloryprow commented:
Can we bump up the walltimes in the drivers and ecf scripts for:

1. **jevs_cam_hireswfv3_grid2obs_stats**: requested walltime is 00:45:00, job ran for 00:43:09
2. **jevs_cam_namnest_grid2obs_stats**: requested walltime 02:15:00, job ran for 02:09:41

AliciaBentley-NOAA commented:
@MarcelCaron-NOAA @malloryprow Some notes I've kept to help with this:

Notes from Wei (NCO SPA: 10/6/23):
runtime 0-2 minutes = minimum 10 minute wall clock
3-10 minutes = minimum 15 minute wall clock
11-15 minutes = minimum 25 minute wall clock
16-20 minutes = minimum 30 minute wall clock
21+ minutes = minimum… runtime * 1.25 rounded up to the nearest 5 min
^^general rules for wallclock limit.

malloryprow commented:
### stats - precip

Submitted jobs with `vhr=21,VDATE=$(date -d "-3 days" +"%Y%m%d")`. COMOUT is /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr597/evs/v2.0/stats/cam

### jevs_cam_hireswarw_precip_stats
**Log File:** /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr597/EVS/dev/drivers/scripts/stats/cam/jevs_cam_hireswarw_precip_stats.o159905784
**DATA:** /lfs/h2/emc/stmp/mallory.row/evs_test/prod/tmp/jevs_cam_hireswarw_precip_stats.159905784.cbqs01

### jevs_cam_hireswarwmem2_precip_stats
**Log File:** /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr597/EVS/dev/drivers/scripts/stats/cam/jevs_cam_hireswarwmem2_precip_stats.o159905787
**DATA:** /lfs/h2/emc/stmp/mallory.row/evs_test/prod/tmp/jevs_cam_hireswarwmem2_precip_stats.159905787.cbqs01

### jevs_cam_hireswfv3_precip_stats
**Log File:** /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr597/EVS/dev/drivers/scripts/stats/cam/jevs_cam_hireswfv3_precip_stats.o159905795
**DATA:** /lfs/h2/emc/stmp/mallory.row/evs_test/prod/tmp/jevs_cam_hireswfv3_precip_stats.159905795.cbqs01

### jevs_cam_hrrr_precip_stats
**Log File:** /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr597/EVS/dev/drivers/scripts/stats/cam/jevs_cam_hrrr_precip_stats.o159905823
**DATA:** /lfs/h2/emc/stmp/mallory.row/evs_test/prod/tmp/jevs_cam_hrrr_precip_stats.159905823.cbqs01

### jevs_cam_namnest_precip_stats
**Log File:** /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr597/EVS/dev/drivers/scripts/stats/cam/jevs_cam_namnest_precip_stats.o159905826
**DATA:** /lfs/h2/emc/stmp/mallory.row/evs_test/prod/tmp/jevs_cam_namnest_precip_stats.159905826.cbqs01

Commit:
Adjust walltimes + match ecf with dev walltimes

MarcelCaron-NOAA commented:
@malloryprow Thanks for checking, walltimes in dev and ecf were increased.

@AliciaBentley-NOAA Thanks for the tip!

Commit:
Match ecf with dev driver walltime

MarcelCaron-NOAA commented:
@malloryprow all precip stats jobs completed cleanly and output look normal:
✔️ jevs_cam_hireswarw_precip_stats
✔️ jevs_cam_hireswarwmem2_precip_stats
✔️ jevs_cam_hireswfv3_precip_stats
✔️ jevs_cam_hrrr_precip_stats
✔️ jevs_cam_namnest_precip_stats

Note:  The recent commit addresses two mismatch ecf walltimes that I missed

malloryprow commented:
Could we get jevs_cam_namnest_precip_stats specifying a memory? I think the ecf scripts list 500 GB. Looking at the last value in the log for `Memory usage` it is ~86 GB. 125GB would be good.

malloryprow commented:
### stats - snowfall

Submitted jobs with `vhr=00`. COMOUT is /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr597/evs/v2.0/stats/cam.

### jevs_cam_hireswarw_snowfall_stats
**Log File:** /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr597/EVS/dev/drivers/scripts/stats/cam/jevs_cam_hireswarw_snowfall_stats.o159992925
**DATA:** /lfs/h2/emc/stmp/mallory.row/evs_test/prod/tmp/jevs_cam_hireswarw_snowfall_stats.159992925.cbqs01

### jevs_cam_hireswarwmem2_snowfall_stats
**Log File:** /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr597/EVS/dev/drivers/scripts/stats/cam/jevs_cam_hireswarwmem2_snowfall_stats.o159992943
**DATA:** /lfs/h2/emc/stmp/mallory.row/evs_test/prod/tmp/jevs_cam_hireswarwmem2_snowfall_stats.159992943.cbqs01

### jevs_cam_hireswfv3_snowfall_stats
**Log File:** /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr597/EVS/dev/drivers/scripts/stats/cam/jevs_cam_hireswfv3_snowfall_stats.o159992945
**DATA:** /lfs/h2/emc/stmp/mallory.row/evs_test/prod/tmp/jevs_cam_hireswfv3_snowfall_stats.159992945.cbqs01

### jevs_cam_hrrr_snowfall_stats
**Log File:** /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr597/EVS/dev/drivers/scripts/stats/cam/jevs_cam_hrrr_snowfall_stats.o159992956
**DATA:** /lfs/h2/emc/stmp/mallory.row/evs_test/prod/tmp/jevs_cam_hrrr_snowfall_stats.159992956.cbqs01

### jevs_cam_namnest_snowfall_stats
**Log File:** /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr597/EVS/dev/drivers/scripts/stats/cam/jevs_cam_namnest_snowfall_stats.o159992970
**DATA:** /lfs/h2/emc/stmp/mallory.row/evs_test/prod/tmp/jevs_cam_namnest_snowfall_stats.159992970.cbqs01

malloryprow commented:
### stats - radar

Submitted jobs with `vhr=00`. COMOUT is /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr597/evs/v2.0/stats/cam.

### jevs_cam_hireswarw_radar_stats
**Log File:** /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr597/EVS/dev/drivers/scripts/stats/cam/jevs_cam_hireswarw_radar_stats.o160002551
**DATA:** /lfs/h2/emc/stmp/mallory.row/evs_test/prod/tmp/jevs_cam_hireswarw_radar_stats.160002551.cbqs01

### jevs_cam_hireswarwmem2_radar_stats
**Log File:** /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr597/EVS/dev/drivers/scripts/stats/cam/jevs_cam_hireswarwmem2_radar_stats.o160002553
**DATA:** /lfs/h2/emc/stmp/mallory.row/evs_test/prod/tmp/jevs_cam_hireswarwmem2_radar_stats.160002553.cbqs01

### jevs_cam_hireswfv3_radar_stats
**Log File:** /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr597/EVS/dev/drivers/scripts/stats/cam/jevs_cam_hireswfv3_radar_stats.o160002554
**DATA:** /lfs/h2/emc/stmp/mallory.row/evs_test/prod/tmp/jevs_cam_hireswfv3_radar_stats.160002554.cbqs01

### jevs_cam_hrrr_radar_stats
**Log File:** /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr597/EVS/dev/drivers/scripts/stats/cam/jevs_cam_hrrr_radar_stats.o160002561
**DATA:** /lfs/h2/emc/stmp/mallory.row/evs_test/prod/tmp/jevs_cam_hrrr_radar_stats.160002561.cbqs01

### jevs_cam_namnest_radar_stats
**Log File:** /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr597/EVS/dev/drivers/scripts/stats/cam/jevs_cam_namnest_radar_stats.o160002563
**DATA:** /lfs/h2/emc/stmp/mallory.row/evs_test/prod/tmp/jevs_cam_namnest_radar_stats.160002563.cbqs01

Commit:
set and adjust mem allocation

Commit:
Match ecf with dev resource requests

MarcelCaron-NOAA commented:
@malloryprow All of the snowfall and radar jobs completed cleanly and output look normal:
✔️ jevs_cam_hireswarw_snowfall_stats
✔️ jevs_cam_hireswarwmem2_snowfall_stats
✔️ jevs_cam_hireswfv3_snowfall_stats
✔️ jevs_cam_hrrr_snowfall_stats
✔️ jevs_cam_namnest_snowfall_stats
✔️ jevs_cam_hireswarw_radar_stats
✔️ jevs_cam_hireswarwmem2_radar_stats
✔️ jevs_cam_hireswfv3_radar_stats
✔️ jevs_cam_hrrr_radar_stats
✔️ jevs_cam_namnest_radar_stats

👍 No other concerns from me about stats jobs tested in this PR

malloryprow commented:
Resources look good for snowfall, but I'd like to adjust the radar ones.

They are all requesting 500 GB of memory so it is taking the whole node for memory but only using npus=3. But the most a job uses is ~21GB. If we set it to 50 GB we can set place=vscatter too and other jobs in the queue can share on the node. I don't want to take the whole node if it isn't needed!

Commit:
Reduce mem allocation

Commit:
Reduce mem allocation (ecf)

MarcelCaron-NOAA commented:
Thanks @malloryprow!  Done 👍

malloryprow commented:
### plots

Submitted jobs with vhr=00. COMOUT is /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr597/evs/v2.0/stats/cam.

### jevs_cam_headline_plots
**Log File:**  /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr597/EVS/dev/drivers/scripts/plots/cam/jevs_cam_headline_plots.o160010034 
**DATA:** /lfs/h2/emc/stmp/mallory.row/evs_test/prod/tmp/jevs_cam_headline_plots.160010034.cbqs01

### jevs_cam_grid2obs_last31days_plots
**Log File:**  /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr597/EVS/dev/drivers/scripts/plots/cam/jevs_cam_grid2obs_plots_last31days.o160010036
**DATA:** /lfs/h2/emc/stmp/mallory.row/evs_test/prod/tmp/jevs_cam_grid2obs_plots_last31days.160010036.cbqs01

### jevs_cam_grid2obs_last90days_plots
**Log File:**  /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr597/EVS/dev/drivers/scripts/plots/cam/jevs_cam_grid2obs_plots_last90days.o160010037
**DATA:** /lfs/h2/emc/stmp/mallory.row/evs_test/prod/tmp/jevs_cam_grid2obs_plots_last90days.160010037.cbqs01

### jevs_cam_precip_last31days_plots
**Log File:**  /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr597/EVS/dev/drivers/scripts/plots/cam/jevs_cam_precip_plots_last31days.o160010039
**DATA:**/lfs/h2/emc/stmp/mallory.row/evs_test/prod/tmp/jevs_cam_precip_plots_last31days.160010039.cbqs01

### jevs_cam_precip_last90days_plots
**Log File:**  /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr597/EVS/dev/drivers/scripts/plots/cam/jevs_cam_precip_plots_last90days.o160010040
**DATA:** /lfs/h2/emc/stmp/mallory.row/evs_test/prod/tmp/jevs_cam_precip_plots_last90days.160010040.cbqs01

### jevs_cam_snowfall_plots
**Log File:**  /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr597/EVS/dev/drivers/scripts/plots/cam/jevs_cam_snowfall_plots.o160010046
**DATA:** /lfs/h2/emc/stmp/mallory.row/evs_test/prod/tmp/jevs_cam_snowfall_plots.160010046.cbqs01

malloryprow commented:
I'm looking  /lfs/h2/emc/stmp/mallory.row/evs_test/prod/tmp/jevs_cam_headline_plots.160010034.cbqs01. Are the jobs each writing a log file to /lfs/h2/emc/stmp/mallory.row/evs_test/prod/tmp/jevs_cam_headline_plots.160010034.cbqs01/headline/out?

malloryprow commented:
> I'm looking /lfs/h2/emc/stmp/mallory.row/evs_test/prod/tmp/jevs_cam_headline_plots.160010034.cbqs01. Are the jobs each writing a log file to /lfs/h2/emc/stmp/mallory.row/evs_test/prod/tmp/jevs_cam_headline_plots.160010034.cbqs01/headline/out?

Oh wait, are they getting copied from their working directory to that location via this? exevs_cam_headline_plots.sh lines 77-81.

```
# Copy Plots Output to Main Directory
for CHILD_DIR in ${DATA}/${VERIF_CASE}/out/workdirs/*; do
    cp -ruv $CHILD_DIR/* ${DATA}/${VERIF_CASE}/out/.
    export err=$?; err_chk
done
```

MarcelCaron-NOAA commented:
@malloryprow yes that's right, that block of code copies the log files and graphics from "workdirs" to the main directory after plotting is complete. 

Child processes write to both $DATA/$VERIF_CASE/out/workdirs/jobX and $DATA/$VERIF_CASE/data/etc/etc/.  The out/ directories are needed in main and are each copied back at the end of all processes via that block of code.  The data/ directories are temp directories and are not copied back.

Commit:
Match ecf with dev driver allocations (plots)

Commit:
Add memory allocation

MarcelCaron-NOAA commented:
@malloryprow The following completed cleanly and output are as expected:
✔️ jevs_cam_headline_plots
✔️ jevs_cam_precip_plots_last31days
✔️ jevs_cam_precip_plots_last90days
✔️ jevs_cam_snowfall_plots

The following are still running:
⏳ jevs_cam_grid2obs_plots_last31days
⏳ jevs_cam_grid2obs_plots_last90days

_Note on recent commits_:  I adjusted some ecf resource configs to match dev.  I also added a mem spec for jevs_cam_headline_plots.

Commit:
Increase mem allocation

MarcelCaron-NOAA commented:
@malloryprow The remaining grid2obs plots jobs have completed cleanly and output files look good:
✔️ jevs_cam_grid2obs_plots_last31days
✔️ jevs_cam_grid2obs_plots_last90days

_Note_: small Cgroup mem limit exceedence on the last31days job.  I increased the memory allocation closer to what is allocated for the last90days job, so it should be good!

👍 no other concerns from me about any of the test jobs listed for this PR

malloryprow commented:
Great news!

Since this PR addresses the NCO MPMD Bugzilla, can you confirm that this fixes that for all the job in https://github.com/NOAA-EMC/EVS/issues/551? If so, that can be closed when this is merged.

AndrewBenjamin-NOAA reviewed (approved):
Barring any additional issues, I approve this PR. Thank you for your diligence in getting this important PR in and for adjusting resources as needed.

MarcelCaron-NOAA commented:
Hi @malloryprow I can confirm this fixes the MPMD Bugzilla in all of the listed jobs except for the following:
```
stats/cam/jevs_cam_hireswarw_severe_stats.sh
stats/cam/jevs_cam_hireswarwmem2_severe_stats.sh
stats/cam/jevs_cam_hireswfv3_severe_stats.sh
stats/cam/jevs_cam_hrrr_severe_stats.sh
stats/cam/jevs_cam_namnest_severe_stats.sh
```

I didn't find that these jobs were running any processes in parallel!

malloryprow commented:
Thanks for checking! We may want to fix the resources on those:

stats/cam/jevs_cam_hireswarw_severe_stats.sh: `select=1:ncpus=5`
stats/cam/jevs_cam_hireswarwmem2_severe_stats.sh: `select=1:ncpus=5`
stats/cam/jevs_cam_hireswfv3_severe_stats.sh: `select=1:ncpus=5`
stats/cam/jevs_cam_hrrr_severe_stats.sh: `select=1:ncpus=5`
stats/cam/jevs_cam_namnest_severe_stats.sh: `select=1:ncpus=5`

They should be `select=1:ncpus=1` if they aren't running anything in parallel.

Commit:
Use single cpu + match ecf with dev resources

malloryprow commented:
With the resource change, I'll want to test those jobs when we get WCOSS2 back.

MarcelCaron-NOAA commented:
Ah right— I've changed the ncpus to one.  

Yes agreed we should test those jobs.  I'm adding them to the PR instructions.  Thanks

malloryprow commented:
### jevs_cam_hireswarw_severe_stats.sh
**Log File:** /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr597/EVS/dev/drivers/scripts/stats/cam/jevs_cam_hireswarw_severe_stats_00.o160997877
**DATA:** /lfs/h2/emc/stmp/mallory.row/evs_test/prod/tmp/jevs_cam_hireswarw_severe_stats_00.160997877.cbqs01

### jevs_cam_hireswarwmem2_severe_stats.sh
**Log File:** /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr597/EVS/dev/drivers/scripts/stats/cam/jevs_cam_hireswarwmem2_severe_stats_00.o160997869
**DATA:**/lfs/h2/emc/stmp/mallory.row/evs_test/prod/tmp/jevs_cam_hireswarwmem2_severe_stats_00.160997869.cbqs01

### jevs_cam_hireswfv3_severe_stats.sh
**Log File:** /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr597/EVS/dev/drivers/scripts/stats/cam/jevs_cam_hireswfv3_severe_stats_00.o160997896
**DATA:** /lfs/h2/emc/stmp/mallory.row/evs_test/prod/tmp/jevs_cam_hireswfv3_severe_stats_00.160997896.cbqs01

### jevs_cam_hrrr_severe_stats.sh
**Log File:** /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr597/EVS/dev/drivers/scripts/stats/cam/jevs_cam_hrrr_severe_stats_00.o160997917
**DATA:** /lfs/h2/emc/stmp/mallory.row/evs_test/prod/tmp/jevs_cam_hrrr_severe_stats_00.160997917.cbqs01

### jevs_cam_namnest_severe_stats.sh
**Log File:** /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr597/EVS/dev/drivers/scripts/stats/cam/jevs_cam_namnest_severe_stats_00.o160997925
**DATA:** /lfs/h2/emc/stmp/mallory.row/evs_test/prod/tmp/jevs_cam_namnest_severe_stats_00.160997925.cbqs01

MarcelCaron-NOAA commented:
Hi @malloryprow.  All jobs ran as expected and output files look normal!
✔️ jevs_cam_hireswarw_severe_stats
✔️ jevs_cam_hireswarwmem2_severe_stats
✔️ jevs_cam_hireswfv3_severe_stats
✔️ jevs_cam_hrrr_severe_stats
✔️ jevs_cam_namnest_severe_stats

malloryprow commented:
Awesome! I believe that this PR is all wrapped up!

malloryprow commented:
Thanks for these changes and working with me on adjustments, @MarcelCaron-NOAA! Cross anything of on the Fixes and Additions document this fixes!
